File: Apps/BankBusiness/frontend/src/api/contact-person.service.ts
import api from './axios';

export interface ContactPersonCreate {
    surname: string;
    name: string;
    patronymic: string;
    phoneNumber: string;
    email: string;
}

export interface ContactPersonRead {
    id: number;
    surname: string;
    name: string;
    patronymic: string;
    phoneNumber: string;
    email: string;
}

export interface ContactPersonUpdate {
    id: number;
    surname: string;
    name: string;
    patronymic: string;
    phoneNumber: string;
    email: string;
}

class ContactPersonService {
    async getAll(): Promise<ContactPersonRead[]> {
        const response = await api.get<ContactPersonRead[]>('/api/contact-person');
        return response.data;
    }

    async getById(id: number): Promise<ContactPersonRead> {
        const response = await api.get<ContactPersonRead>(`/api/contact-person?id=${id}`);
        return response.data;
    }

    async create(data: ContactPersonCreate): Promise<void> {
        await api.post('/api/contact-person', data);
    }

    async update(data: ContactPersonUpdate): Promise<void> {
        await api.put('/api/contact-person', data);
    }

    async delete(personId: number): Promise<void> {
        await api.delete(`/api/contact-person?personId=${personId}`);
    }
}

export default new ContactPersonService();

File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/services/business/ContactPersonService.java
package org.d3javu.backend.services.business;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.d3javu.backend.kafka.main.business.contactperson.ContactPersonCreateRequest;
import org.d3javu.backend.kafka.main.business.contactperson.ContactPersonDeleteRequest;
import org.d3javu.backend.kafka.main.business.contactperson.ContactPersonUpdateRequest;
import org.d3javu.backend.repository.business.ContactPersonRepository;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class ContactPersonService {

    private final ContactPersonRepository contactPersonRepository;
    private final BusinessClientService businessClientService;

    public void create(ContactPersonCreateRequest request) {
        this.businessClientService
                .getIdByEmail(request.businessClientEmail())
                .ifPresentOrElse(id -> this.contactPersonRepository.create(
                        id,
                        request.surname(),
                        request.name(),
                        request.patronymic(),
                        request.phoneNumber(),
                        request.email()
                ), () -> {
                    log.warn("not found id for business client {}", request.businessClientEmail());
                });
    }

    public void update(ContactPersonUpdateRequest request) {
        this.businessClientService
                .getIdByEmail(request.businessClientEmail())
                .ifPresentOrElse(id -> {try {
                    this.contactPersonRepository.update(
                    id,
                    request.contactPersonId(),
                    request.surname(),
                    request.name(),
                    request.patronymic(),
                    request.phoneNumber(),
                    request.email());
                } catch (Exception e) {
                    log.error("error {}", e);
                }
        }, () -> log.warn("not found id for business client {}", request.businessClientEmail()));
    }

    public void delete(ContactPersonDeleteRequest request) {
        this.businessClientService
                .getIdByEmail(request.businessClientEmail())
                .ifPresentOrElse(id -> this.contactPersonRepository.delete(
                        id,
                        request.personId()
                ), () -> log.warn("not found id for business client {}", request.businessClientEmail()));
    }
}
